Marlin/Marlin_main.cpp

```diff
@@ -9737,7 +9737,9 @@ void setup() {
       safe_delay(BOOTSCREEN_TIMEOUT);
     #elif ENABLED(ULTRA_LCD)
       bootscreen();
-      lcd_init();
+      #if DISABLED(SDSUPPORT)
+        lcd_init();
+      #endif
```

Comment:
This second `lcd_init()` causes freeze when `SDSUPPORT` (and MAX31855) is enabled for some reason.
But when `SDSUPPORT` is disabled, this doesn't causes freeze, moreover if this second `lcd_init()` is removed, displaying of LCD is broken.